Amplify/Categories/DataStore/Model/Model.swift

```diff
@@ -30,3 +30,16 @@ public protocol Model: Codable {
     var id: Identifier { get }
 
 }
+
+public protocol JSONModel: Model {
+
+    /// Return the value for key
+    func internalValue(for key: String) -> Any?
```

Comment:
1. As discussed, let's rename this method to `jsonValue(forKey:)`
1. As discussed, this should be returning a `JSONValue?` not an `Any?`

Reply:
1. Renamed
2. Returns an Any??,  should we enforce the concrete class to use `JSONValue` as its internal implementation? They can use their own implementation of JSONValue right? Any gives them the flexibility?